packages/flutter/lib/src/services/deferred_component.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+import 'system_channels.dart';
+
+/// Default Google Play Store implementation states that reflect the current
+/// installation status of a deferred component module.
+///
+/// Returned by [DeferredComponent.getDeferredComponentInstallState].
+///
+/// These states may differ in custom deferred component implementations.
+///
+/// Due to the asynchronous nature of platform channels, these states are
```

Comment:
This paragraph makes me wonder what the actual use case for this API is if I can't depend on it. Why would a developer call it?

Reply:
The intended use of this is to display loading messages/bars/screens/etc, though it can also be for stuff like logging.

Reply:
Maybe mention this as examples in the docs?